src/backgrounds.jl

```diff
@@ -0,0 +1,194 @@
+"""
+
+`draw_grid(video::Video, action::Action, frame::Int; direction::String = "TR", line_gap::Int = 25)`
+
+Draws an oriented grid on the given frame of a Video.
+
+# Arguments
+- `direction::String`: Where grid animation finishes. Default: `"TR"` Available Orientations:
+  - `"TR"` - Animation finishes in the **T**op **R**ight corner of the frame.
+  - `"TL"` - Animation finishes in the **T**op **L**eft corner of the frame.
+  - `"BR"` - Animation finishes in the **B**ottom **R**ight corner of the frame.
+  - `"BL"` - Animation finishes in the **B**ottom **L**eft corner of the frame.
+- `line_gap::Int`: How many pixels between each line. Default: `25`
+
+# Example
+Example call of this function within an `Action`.
+```
+...
+ Action(1:100, :line, draw_grid(direction = "TL", line_gap = 25))
+...
+```
+
+"""
+function draw_grid(
+    video::Video,
+    action::Action,
+    frame::Int;
+    direction::String = "TR",
+    line_gap::Int = 25,
+)
+
+    min_width = div(video.width, -2, RoundDown)
+    max_width = div(video.width, 2, RoundUp)
+
+    min_height = div(video.height, -2, RoundDown)
+    max_height = div(video.height, 2, RoundUp)
+
+    step = (frame - first(action.frames)) / (length(action.frames) - 1)
+    if occursin("B", direction)
+
+        for x_point = min_height:line_gap:max_height
+            start_point = Point(x_point, min_height)
+            end_point = start_point + step * (Point(x_point, max_height) - start_point)
+            line(start_point, end_point, :stroke)
+        end
+
+        if occursin("R", direction)
+
+            for y_point = min_width:line_gap:max_width
+                start_point = Point(min_width, y_point)
+                end_point = start_point + step * (Point(max_width, y_point) - start_point)
+                line(start_point, end_point, :stroke)
+            end
+
+        else
+
+            for y_point = min_width:line_gap:max_width
+                start_point = Point(max_width, y_point)
+                end_point = start_point + step * (Point(min_width, y_point) - start_point)
+                line(start_point, end_point, :stroke)
+            end
+
+        end
+
+    else
+
+        for x_point = min_height:line_gap:max_height
+            start_point = Point(x_point, max_height)
+            end_point = start_point + step * (Point(x_point, min_height) - start_point)
+            line(start_point, end_point, :stroke)
+        end
+
+        if occursin("R", direction)
```

Comment:
Isn't this whole block the same as in the one under `occursin("B"` ? Additionally `occursin` is probably not the best way to handle this. You might want to ask either `if direction == "TR" || direction == "BR"` which is the most readable or 
`if direction[2] == "R"`

Reply:
We got this sorted out. :smile: